---
author: "Khunanon Chanasongkhram"
date: "14/08/2023"
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# DisVar: An R library for identifying variants associated with diseases using personal genetic information

<!-- badges: start -->
<!-- badges: end -->
DisVar is an R package that finds diseases from a VCF file by comparing the variants in the VCF file with those in various databases. The package currently supports five databases: GWASdb, GRASP, GWAS Catalog, GAD, and Johnson and O'Donnell.

## Features

- Finds diseases from a VCF file by comparing the variants in the VCF file with those in various databases.

- Supports five databases: GWASdb, GRASP, GWAS Catalog, GAD, and Johnson and O'Donnell.

## Installation

## Installing the Package in R Studio
You can install the development version of DisVar from [GitHub](https://github.com/). To install the package, you can use the `devtools` package and run the following command:

``` r
# install.packages("devtools")
devtools::install_github("Khunanon-Chanasongkhram/DisVar")
```

## Installing the Package on Linux servers

### Step 1: Installing Minoconda 3

To install Miniconda via the terminal, you can follow these steps:
1.Create a directory to install Miniconda into:

``` 
mkdir -p ~/miniconda3
```

2.Download the latest Miniconda3 based install script for Linux 64 bit:
``` 
wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda3/miniconda.sh
```

3.Run the install script:
``` 
bash ~/miniconda3/miniconda.sh -b -u -p ~/miniconda3
```

After then, you can restart your shell, and conda will be ready for use.

For further details regarding conda installation please see [Miniconda linux installers](https://docs.conda.io/en/latest/miniconda.html#linux-installers)

### Step 2: Installing R using conda
Create a new conda environment for R by running the following command:
``` 
conda create -n r-environment r-base
```

Please answer "y" for the question "Proceed ([y]/n)?".
This will create a new environment named 'r-environment' with R installed.
Now you can activate the new environment by running:
```
conda activate r-environment
```

If you want to and deactivate an active environment just running:

```
conda deactivate
```

### Step 3: Installing the DisVar Package
After activate r-environment via conda. Run R in Linux by following command:
```
R
```

Then install the package
```
remotes::install_github(repo = "Khunanon-Chanasongkhram/DisVar",dependencies = TRUE,upgrade = TRUE)
```
Now the DisVar is ready to use!

## Dependencies
The package depends on the following packages:

- `sqldf`
- `data.table`
- `dplyr`

## Usage

The main function in the package is `DisVar()`, which takes one arguments:

- `file`: the name of the VCF file.

The function returns a data frame containing variants associated with diseases compared from various databases.

### Examples

Here's an example of how to use the function:

```{r example}
library(DisVar)
DisVar("file_name.vcf")
```

You will get this output file: `file_name`_DisVar.txt

When you test the program with the test file (`file_name.vcf`), the results are shown in the following table.

| Disease                                                    | Chrom | Position | Gene    | Variant ID | Variant Type     | Allele Sample | Allele DB | P-value   | DB           | Qual    | Filter | Info                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |
|------------------------------------------------------------|-------|----------|---------|------------|------------------|---------------|-----------|-----------|--------------|---------|--------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Alzheimer's disease                                        | 6     | 41161514 | TREM2   | rs75932628 | missense         | C>T           | C>T       | 2.00E-12  | GWASdb       | 7296.45 | PASS   | AC=14;AF=0.00218613;AN=6404;BaseQRankSum=0.155;ClippingRankSum=0.481;DP=111734;FS=1.098;MLEAC=14;MLEAF=0.002186;MQ=60;MQ0=0;MQRankSum=0.307;POSITIVE_TRAIN_SITE;QD=13.46;ReadPosRankSum=0.839;SOR=0.59;VQSLOD=16.39;culprit=MQ;AN_EUR=1266;AN_EAS=1170;AN_AMR=980;AN_SAS=1202;AN_AFR=1786;AF_EUR=0.00631912;AF_EAS=0;AF_AMR=0.00306122;AF_SAS=0.00249584;AF_AFR=0;AC_EUR=8;AC_EAS=0;AC_AMR=3;AC_SAS=3;AC_AFR=0;AC_Het_EUR=8;AC_Het_EAS=0;AC_Het_AMR=3;AC_Het_SAS=3;AC_Het_AFR=0;AC_Het=14;AC_Hom_EUR=0;AC_Hom_EAS=0;AC_Hom_AMR=0;AC_Hom_SAS=0;AC_Hom_AFR=0;AC_Hom=0;HWE_EUR=1;ExcHet_EUR=0.977997;HWE_EAS=1;ExcHet_EAS=1;HWE_AMR=1;ExcHet_AMR=0.996936;HWE_SAS=1;ExcHet_SAS=0.997502;HWE_AFR=1;ExcHet_AFR=1;HWE=1;ExcHet=0.985861;ME=0;AN_EUR_unrel=1006;AN_EAS_unrel=1008;AN_AMR_unrel=694;AN_SAS_unrel=978;AN_AFR_unrel=1322;AF_EUR_unrel=0.00497018;AF_EAS_unrel=0;AF_AMR_unrel=0.00432277;AF_SAS_unrel=0.00204499;AF_AFR_unrel=0;AC_EUR_unrel=5;AC_EAS_unrel=0;AC_AMR_unrel=3;AC_SAS_unrel=2;AC_AFR_unrel=0;AC_Het_EUR_unrel=5;AC_Het_EAS_unrel=0;AC_Het_AMR_unrel=3;AC_Het_SAS_unrel=2;AC_Het_AFR_unrel=0;AC_Hom_EUR_unrel=0;AC_Hom_EAS_unrel=0;AC_Hom_AMR_unrel=0;AC_Hom_SAS_unrel=0;AC_Hom_AFR_unrel=0;ANN=T\|missense_variant\|MODERATE\|TREM2\|ENSG00000095970\|transcript\|ENST00000373113.7\|protein_coding\|2/5\|c.140G>A\|p.Arg47His\|234/1044\|140/693\|47/230\|\|;1000Gp3_RA_GF=0.00399361;1000Gp3_RR_GF=0.996006;1000Gp3_HomC=2494,0;1000Gp3_AA_GF=0;1000Gp3_AF=0.00199681;CLNSIG=3;CLNDBN=not_specified;CLNDSDBID=CN169374;CLNDSDB=MedGen;COSMIC_CNT=1;SIFT_pred=D/T/T;Uniprot_id_Polyphen2=././TREM2_HUMAN;Uniprot_aapos_Polyphen2=47;Polyphen2_HDIV_pred=D;Polyphen2_HVAR_pred=D;MutationTaster_pred=D/D/D;MutationAssessor_pred=M;CADD_phred=33;GERP++_NR=5.51;GERP++_RS=4.56;phyloP20way_mammalian=0.935;phastCons20way_mammalian=0.997;Entrez_gene_id=54209;RVIS=0.483275131;RVIS_percentile=79.25218212;GDI=446.17961;GDI-Phred=4.39760;LoFtool_score=8.38E-01;Essential_gene=N;MGI_mouse_gene=Trem2;dbSNPBuildID=132;ExAC_AF=0.002043 |
| Alzheimer's disease (late onset)                           | 8     | 95041772 | C8orf38 | rs7818382  | intron           | C>T           | C>T       | 8.00E-08  | GWASdb       | 1587730 | Pass   | AC=3017;AF=0.471112;AN=6404;BaseQRankSum=0.223;ClippingRankSum=0;DP=106075;FS=0;InbreedingCoeff=0.1044;MLEAC=3017;MLEAF=0.471;MQ=60;MQ0=0;MQRankSum=-0.022;POSITIVE_TRAIN_SITE;QD=21.26;ReadPosRankSum=0.319;SOR=0.686;VQSLOD=23.35;culprit=MQ;AN_EUR=1266;AN_EAS=1170;AN_AMR=980;AN_SAS=1202;AN_AFR=1786;AF_EUR=0.470774;AF_EAS=0.670085;AF_AMR=0.529592;AF_SAS=0.569884;AF_AFR=0.242441;AC_EUR=596;AC_EAS=784;AC_AMR=519;AC_SAS=685;AC_AFR=433;AC_Het_EUR=320;AC_Het_EAS=240;AC_Het_AMR=237;AC_Het_SAS=295;AC_Het_AFR=337;AC_Het=1429;AC_Hom_EUR=276;AC_Hom_EAS=544;AC_Hom_AMR=282;AC_Hom_SAS=390;AC_Hom_AFR=96;AC_Hom=1588;HWE_EUR=0.749959;ExcHet_EUR=0.395209;HWE_EAS=0.0919533;ExcHet_EAS=0.968002;HWE_AMR=0.526092;ExcHet_AMR=0.776731;HWE_SAS=1;ExcHet_SAS=0.529247;HWE_AFR=0.466134;ExcHet_AFR=0.239997;HWE=3.8034e-09;ExcHet=1;ME=0;AN_EUR_unrel=1006;AN_EAS_unrel=1008;AN_AMR_unrel=694;AN_SAS_unrel=978;AN_AFR_unrel=1322;AF_EUR_unrel=0.464215;AF_EAS_unrel=0.686508;AF_AMR_unrel=0.51585;AF_SAS_unrel=0.566462;AF_AFR_unrel=0.249622;AC_EUR_unrel=467;AC_EAS_unrel=692;AC_AMR_unrel=358;AC_SAS_unrel=554;AC_AFR_unrel=330;AC_Het_EUR_unrel=253;AC_Het_EAS_unrel=200;AC_Het_AMR_unrel=162;AC_Het_SAS_unrel=238;AC_Het_AFR_unrel=256;AC_Hom_EUR_unrel=214;AC_Hom_EAS_unrel=492;AC_Hom_AMR_unrel=196;AC_Hom_SAS_unrel=316;AC_Hom_AFR_unrel=74;ANN=T\|intron_variant\|MODIFIER\|NDUFAF6\|ENSG00000156170\|transcript\|ENST00000396124.8\|protein_coding\|4/8\|c.477+146C>T\|\|\|\|\|\|;1000Gp3_RA_GF=0.442891;1000Gp3_RR_GF=0.299121;1000Gp3_HomC=749,646;1000Gp3_AA_GF=0.257987;1000Gp3_AF=0.479433;Entrez_gene_id=137682;RVIS=-0.161524709;RVIS_percentile=41.6430762;GDI=43.71035;GDI-Phred=1.25843;MGI_mouse_gene=Ndufaf6;dbSNPBuildID=116                                                                                                                                                                                                                                                                                                      |
| Breast cancer                                              | 10    | 1.22E+08 | NA      | rs2981579  | NA               | A>G           | NA        | 1.90E-170 | GRASP        | 1491780 | PASS   | AC=3227;AF=0.503904;AN=6404;BaseQRankSum=0.098;ClippingRankSum=-0.024;DP=95506;FS=0;InbreedingCoeff=0.0499;MLEAC=3227;MLEAF=0.504;MQ=60;MQ0=0;MQRankSum=0.027;POSITIVE_TRAIN_SITE;QD=21.12;ReadPosRankSum=0.349;SOR=0.676;VQSLOD=22.69;culprit=MQ;AN_EUR=1266;AN_EAS=1170;AN_AMR=980;AN_SAS=1202;AN_AFR=1786;AF_EUR=0.550553;AF_EAS=0.549573;AF_AMR=0.568367;AF_SAS=0.610649;AF_AFR=0.333707;AC_EUR=697;AC_EAS=643;AC_AMR=557;AC_SAS=734;AC_AFR=596;AC_Het_EUR=293;AC_Het_EAS=283;AC_Het_AMR=249;AC_Het_SAS=296;AC_Het_AFR=400;AC_Het=1521;AC_Hom_EUR=404;AC_Hom_EAS=360;AC_Hom_AMR=308;AC_Hom_SAS=438;AC_Hom_AFR=196;AC_Hom=1706;HWE_EUR=0.108258;ExcHet_EUR=0.95804;HWE_EAS=0.616227;ExcHet_EAS=0.744677;HWE_AMR=0.462052;ExcHet_AMR=0.249731;HWE_SAS=0.439899;ExcHet_SAS=0.220669;HWE_AFR=0.880403;ExcHet_AFR=0.45117;HWE=0.00469594;ExcHet=0.997943;ME=0;AN_EUR_unrel=1006;AN_EAS_unrel=1008;AN_AMR_unrel=694;AN_SAS_unrel=978;AN_AFR_unrel=1322;AF_EUR_unrel=0.548708;AF_EAS_unrel=0.550595;AF_AMR_unrel=0.573487;AF_SAS_unrel=0.611452;AF_AFR_unrel=0.33888;AC_EUR_unrel=552;AC_EAS_unrel=555;AC_AMR_unrel=398;AC_SAS_unrel=598;AC_AFR_unrel=448;AC_Het_EUR_unrel=228;AC_Het_EAS_unrel=251;AC_Het_AMR_unrel=176;AC_Het_SAS_unrel=234;AC_Het_AFR_unrel=296;AC_Hom_EUR_unrel=324;AC_Hom_EAS_unrel=304;AC_Hom_AMR_unrel=222;AC_Hom_SAS_unrel=364;AC_Hom_AFR_unrel=152;ANN=G\|intron_variant\|MODIFIER\|FGFR2\|ENSG00000066468\|transcript\|ENST00000457416.6\|protein_coding\|2/17\|c.110-12117T>C\|\|\|\|\|\|;1000Gp3_RA_GF=0.473243;1000Gp3_RR_GF=0.253994;1000Gp3_HomC=636,683;1000Gp3_AA_GF=0.272764;1000Gp3_AF=0.509385;AR_GENE=FGFR2;Entrez_gene_id=2263;RVIS=-1.50462206;RVIS_percentile=3.568058504;GDI=593.43974;GDI-Phred=4.93642;LoFtool_score=1.79E-03;Essential_gene=E;MGI_mouse_gene=Fgfr2;ZFIN_zebrafish_gene=fgfr2;ZFIN_zebrafish_phenotype_tag=abnormal;dbSNPBuildID=101                                                                                                                                                                  |
|                                                            | 10    | 1.22E+08 | FGFR2   | rs2981579  | intron           | A>G           | A>G       | 2.00E-170 | GWASdb       | 1491780 | PASS   | AC=3227;AF=0.503904;AN=6404;BaseQRankSum=0.098;ClippingRankSum=-0.024;DP=95506;FS=0;InbreedingCoeff=0.0499;MLEAC=3227;MLEAF=0.504;MQ=60;MQ0=0;MQRankSum=0.027;POSITIVE_TRAIN_SITE;QD=21.12;ReadPosRankSum=0.349;SOR=0.676;VQSLOD=22.69;culprit=MQ;AN_EUR=1266;AN_EAS=1170;AN_AMR=980;AN_SAS=1202;AN_AFR=1786;AF_EUR=0.550553;AF_EAS=0.549573;AF_AMR=0.568367;AF_SAS=0.610649;AF_AFR=0.333707;AC_EUR=697;AC_EAS=643;AC_AMR=557;AC_SAS=734;AC_AFR=596;AC_Het_EUR=293;AC_Het_EAS=283;AC_Het_AMR=249;AC_Het_SAS=296;AC_Het_AFR=400;AC_Het=1521;AC_Hom_EUR=404;AC_Hom_EAS=360;AC_Hom_AMR=308;AC_Hom_SAS=438;AC_Hom_AFR=196;AC_Hom=1706;HWE_EUR=0.108258;ExcHet_EUR=0.95804;HWE_EAS=0.616227;ExcHet_EAS=0.744677;HWE_AMR=0.462052;ExcHet_AMR=0.249731;HWE_SAS=0.439899;ExcHet_SAS=0.220669;HWE_AFR=0.880403;ExcHet_AFR=0.45117;HWE=0.00469594;ExcHet=0.997943;ME=0;AN_EUR_unrel=1006;AN_EAS_unrel=1008;AN_AMR_unrel=694;AN_SAS_unrel=978;AN_AFR_unrel=1322;AF_EUR_unrel=0.548708;AF_EAS_unrel=0.550595;AF_AMR_unrel=0.573487;AF_SAS_unrel=0.611452;AF_AFR_unrel=0.33888;AC_EUR_unrel=552;AC_EAS_unrel=555;AC_AMR_unrel=398;AC_SAS_unrel=598;AC_AFR_unrel=448;AC_Het_EUR_unrel=228;AC_Het_EAS_unrel=251;AC_Het_AMR_unrel=176;AC_Het_SAS_unrel=234;AC_Het_AFR_unrel=296;AC_Hom_EUR_unrel=324;AC_Hom_EAS_unrel=304;AC_Hom_AMR_unrel=222;AC_Hom_SAS_unrel=364;AC_Hom_AFR_unrel=152;ANN=G\|intron_variant\|MODIFIER\|FGFR2\|ENSG00000066468\|transcript\|ENST00000457416.6\|protein_coding\|2/17\|c.110-12117T>C\|\|\|\|\|\|;1000Gp3_RA_GF=0.473243;1000Gp3_RR_GF=0.253994;1000Gp3_HomC=636,683;1000Gp3_AA_GF=0.272764;1000Gp3_AF=0.509385;AR_GENE=FGFR2;Entrez_gene_id=2263;RVIS=-1.50462206;RVIS_percentile=3.568058504;GDI=593.43974;GDI-Phred=4.93642;LoFtool_score=1.79E-03;Essential_gene=E;MGI_mouse_gene=Fgfr2;ZFIN_zebrafish_gene=fgfr2;ZFIN_zebrafish_phenotype_tag=abnormal;dbSNPBuildID=101                                                                                                                                                                  |
|                                                            | 10    | 1.22E+08 | FGFR2   | rs2981579  | intron           | A>G           | A>G       | 4.00E-31  | GWASdb       | 1491780 | PASS   | AC=3227;AF=0.503904;AN=6404;BaseQRankSum=0.098;ClippingRankSum=-0.024;DP=95506;FS=0;InbreedingCoeff=0.0499;MLEAC=3227;MLEAF=0.504;MQ=60;MQ0=0;MQRankSum=0.027;POSITIVE_TRAIN_SITE;QD=21.12;ReadPosRankSum=0.349;SOR=0.676;VQSLOD=22.69;culprit=MQ;AN_EUR=1266;AN_EAS=1170;AN_AMR=980;AN_SAS=1202;AN_AFR=1786;AF_EUR=0.550553;AF_EAS=0.549573;AF_AMR=0.568367;AF_SAS=0.610649;AF_AFR=0.333707;AC_EUR=697;AC_EAS=643;AC_AMR=557;AC_SAS=734;AC_AFR=596;AC_Het_EUR=293;AC_Het_EAS=283;AC_Het_AMR=249;AC_Het_SAS=296;AC_Het_AFR=400;AC_Het=1521;AC_Hom_EUR=404;AC_Hom_EAS=360;AC_Hom_AMR=308;AC_Hom_SAS=438;AC_Hom_AFR=196;AC_Hom=1706;HWE_EUR=0.108258;ExcHet_EUR=0.95804;HWE_EAS=0.616227;ExcHet_EAS=0.744677;HWE_AMR=0.462052;ExcHet_AMR=0.249731;HWE_SAS=0.439899;ExcHet_SAS=0.220669;HWE_AFR=0.880403;ExcHet_AFR=0.45117;HWE=0.00469594;ExcHet=0.997943;ME=0;AN_EUR_unrel=1006;AN_EAS_unrel=1008;AN_AMR_unrel=694;AN_SAS_unrel=978;AN_AFR_unrel=1322;AF_EUR_unrel=0.548708;AF_EAS_unrel=0.550595;AF_AMR_unrel=0.573487;AF_SAS_unrel=0.611452;AF_AFR_unrel=0.33888;AC_EUR_unrel=552;AC_EAS_unrel=555;AC_AMR_unrel=398;AC_SAS_unrel=598;AC_AFR_unrel=448;AC_Het_EUR_unrel=228;AC_Het_EAS_unrel=251;AC_Het_AMR_unrel=176;AC_Het_SAS_unrel=234;AC_Het_AFR_unrel=296;AC_Hom_EUR_unrel=324;AC_Hom_EAS_unrel=304;AC_Hom_AMR_unrel=222;AC_Hom_SAS_unrel=364;AC_Hom_AFR_unrel=152;ANN=G\|intron_variant\|MODIFIER\|FGFR2\|ENSG00000066468\|transcript\|ENST00000457416.6\|protein_coding\|2/17\|c.110-12117T>C\|\|\|\|\|\|;1000Gp3_RA_GF=0.473243;1000Gp3_RR_GF=0.253994;1000Gp3_HomC=636,683;1000Gp3_AA_GF=0.272764;1000Gp3_AF=0.509385;AR_GENE=FGFR2;Entrez_gene_id=2263;RVIS=-1.50462206;RVIS_percentile=3.568058504;GDI=593.43974;GDI-Phred=4.93642;LoFtool_score=1.79E-03;Essential_gene=E;MGI_mouse_gene=Fgfr2;ZFIN_zebrafish_gene=fgfr2;ZFIN_zebrafish_phenotype_tag=abnormal;dbSNPBuildID=101                                                                                                                                                                  |
|                                                            | 10    | 1.22E+08 | FGFR2   | rs2981579  | intron           | A>G           | A>G       | 2.00E-10  | GWASdb       | 1491780 | PASS   | AC=3227;AF=0.503904;AN=6404;BaseQRankSum=0.098;ClippingRankSum=-0.024;DP=95506;FS=0;InbreedingCoeff=0.0499;MLEAC=3227;MLEAF=0.504;MQ=60;MQ0=0;MQRankSum=0.027;POSITIVE_TRAIN_SITE;QD=21.12;ReadPosRankSum=0.349;SOR=0.676;VQSLOD=22.69;culprit=MQ;AN_EUR=1266;AN_EAS=1170;AN_AMR=980;AN_SAS=1202;AN_AFR=1786;AF_EUR=0.550553;AF_EAS=0.549573;AF_AMR=0.568367;AF_SAS=0.610649;AF_AFR=0.333707;AC_EUR=697;AC_EAS=643;AC_AMR=557;AC_SAS=734;AC_AFR=596;AC_Het_EUR=293;AC_Het_EAS=283;AC_Het_AMR=249;AC_Het_SAS=296;AC_Het_AFR=400;AC_Het=1521;AC_Hom_EUR=404;AC_Hom_EAS=360;AC_Hom_AMR=308;AC_Hom_SAS=438;AC_Hom_AFR=196;AC_Hom=1706;HWE_EUR=0.108258;ExcHet_EUR=0.95804;HWE_EAS=0.616227;ExcHet_EAS=0.744677;HWE_AMR=0.462052;ExcHet_AMR=0.249731;HWE_SAS=0.439899;ExcHet_SAS=0.220669;HWE_AFR=0.880403;ExcHet_AFR=0.45117;HWE=0.00469594;ExcHet=0.997943;ME=0;AN_EUR_unrel=1006;AN_EAS_unrel=1008;AN_AMR_unrel=694;AN_SAS_unrel=978;AN_AFR_unrel=1322;AF_EUR_unrel=0.548708;AF_EAS_unrel=0.550595;AF_AMR_unrel=0.573487;AF_SAS_unrel=0.611452;AF_AFR_unrel=0.33888;AC_EUR_unrel=552;AC_EAS_unrel=555;AC_AMR_unrel=398;AC_SAS_unrel=598;AC_AFR_unrel=448;AC_Het_EUR_unrel=228;AC_Het_EAS_unrel=251;AC_Het_AMR_unrel=176;AC_Het_SAS_unrel=234;AC_Het_AFR_unrel=296;AC_Hom_EUR_unrel=324;AC_Hom_EAS_unrel=304;AC_Hom_AMR_unrel=222;AC_Hom_SAS_unrel=364;AC_Hom_AFR_unrel=152;ANN=G\|intron_variant\|MODIFIER\|FGFR2\|ENSG00000066468\|transcript\|ENST00000457416.6\|protein_coding\|2/17\|c.110-12117T>C\|\|\|\|\|\|;1000Gp3_RA_GF=0.473243;1000Gp3_RR_GF=0.253994;1000Gp3_HomC=636,683;1000Gp3_AA_GF=0.272764;1000Gp3_AF=0.509385;AR_GENE=FGFR2;Entrez_gene_id=2263;RVIS=-1.50462206;RVIS_percentile=3.568058504;GDI=593.43974;GDI-Phred=4.93642;LoFtool_score=1.79E-03;Essential_gene=E;MGI_mouse_gene=Fgfr2;ZFIN_zebrafish_gene=fgfr2;ZFIN_zebrafish_phenotype_tag=abnormal;dbSNPBuildID=101                                                                                                                                                                  |
|                                                            | 10    | 1.22E+08 | FGFR2   | rs2981579  | NA               | A>G           | NA        | NA        | GADCDC       | 1491780 | PASS   | AC=3227;AF=0.503904;AN=6404;BaseQRankSum=0.098;ClippingRankSum=-0.024;DP=95506;FS=0;InbreedingCoeff=0.0499;MLEAC=3227;MLEAF=0.504;MQ=60;MQ0=0;MQRankSum=0.027;POSITIVE_TRAIN_SITE;QD=21.12;ReadPosRankSum=0.349;SOR=0.676;VQSLOD=22.69;culprit=MQ;AN_EUR=1266;AN_EAS=1170;AN_AMR=980;AN_SAS=1202;AN_AFR=1786;AF_EUR=0.550553;AF_EAS=0.549573;AF_AMR=0.568367;AF_SAS=0.610649;AF_AFR=0.333707;AC_EUR=697;AC_EAS=643;AC_AMR=557;AC_SAS=734;AC_AFR=596;AC_Het_EUR=293;AC_Het_EAS=283;AC_Het_AMR=249;AC_Het_SAS=296;AC_Het_AFR=400;AC_Het=1521;AC_Hom_EUR=404;AC_Hom_EAS=360;AC_Hom_AMR=308;AC_Hom_SAS=438;AC_Hom_AFR=196;AC_Hom=1706;HWE_EUR=0.108258;ExcHet_EUR=0.95804;HWE_EAS=0.616227;ExcHet_EAS=0.744677;HWE_AMR=0.462052;ExcHet_AMR=0.249731;HWE_SAS=0.439899;ExcHet_SAS=0.220669;HWE_AFR=0.880403;ExcHet_AFR=0.45117;HWE=0.00469594;ExcHet=0.997943;ME=0;AN_EUR_unrel=1006;AN_EAS_unrel=1008;AN_AMR_unrel=694;AN_SAS_unrel=978;AN_AFR_unrel=1322;AF_EUR_unrel=0.548708;AF_EAS_unrel=0.550595;AF_AMR_unrel=0.573487;AF_SAS_unrel=0.611452;AF_AFR_unrel=0.33888;AC_EUR_unrel=552;AC_EAS_unrel=555;AC_AMR_unrel=398;AC_SAS_unrel=598;AC_AFR_unrel=448;AC_Het_EUR_unrel=228;AC_Het_EAS_unrel=251;AC_Het_AMR_unrel=176;AC_Het_SAS_unrel=234;AC_Het_AFR_unrel=296;AC_Hom_EUR_unrel=324;AC_Hom_EAS_unrel=304;AC_Hom_AMR_unrel=222;AC_Hom_SAS_unrel=364;AC_Hom_AFR_unrel=152;ANN=G\|intron_variant\|MODIFIER\|FGFR2\|ENSG00000066468\|transcript\|ENST00000457416.6\|protein_coding\|2/17\|c.110-12117T>C\|\|\|\|\|\|;1000Gp3_RA_GF=0.473243;1000Gp3_RR_GF=0.253994;1000Gp3_HomC=636,683;1000Gp3_AA_GF=0.272764;1000Gp3_AF=0.509385;AR_GENE=FGFR2;Entrez_gene_id=2263;RVIS=-1.50462206;RVIS_percentile=3.568058504;GDI=593.43974;GDI-Phred=4.93642;LoFtool_score=1.79E-03;Essential_gene=E;MGI_mouse_gene=Fgfr2;ZFIN_zebrafish_gene=fgfr2;ZFIN_zebrafish_phenotype_tag=abnormal;dbSNPBuildID=101                                                                                                                                                                  |
| Nasopharyngeal carcinoma                                   | 10    | 1.22E+08 | FGFR2   | rs2981579  | intron           | A>G           | A>G       | 2.00E-10  | GWASdb       | 1491780 | PASS   | AC=3227;AF=0.503904;AN=6404;BaseQRankSum=0.098;ClippingRankSum=-0.024;DP=95506;FS=0;InbreedingCoeff=0.0499;MLEAC=3227;MLEAF=0.504;MQ=60;MQ0=0;MQRankSum=0.027;POSITIVE_TRAIN_SITE;QD=21.12;ReadPosRankSum=0.349;SOR=0.676;VQSLOD=22.69;culprit=MQ;AN_EUR=1266;AN_EAS=1170;AN_AMR=980;AN_SAS=1202;AN_AFR=1786;AF_EUR=0.550553;AF_EAS=0.549573;AF_AMR=0.568367;AF_SAS=0.610649;AF_AFR=0.333707;AC_EUR=697;AC_EAS=643;AC_AMR=557;AC_SAS=734;AC_AFR=596;AC_Het_EUR=293;AC_Het_EAS=283;AC_Het_AMR=249;AC_Het_SAS=296;AC_Het_AFR=400;AC_Het=1521;AC_Hom_EUR=404;AC_Hom_EAS=360;AC_Hom_AMR=308;AC_Hom_SAS=438;AC_Hom_AFR=196;AC_Hom=1706;HWE_EUR=0.108258;ExcHet_EUR=0.95804;HWE_EAS=0.616227;ExcHet_EAS=0.744677;HWE_AMR=0.462052;ExcHet_AMR=0.249731;HWE_SAS=0.439899;ExcHet_SAS=0.220669;HWE_AFR=0.880403;ExcHet_AFR=0.45117;HWE=0.00469594;ExcHet=0.997943;ME=0;AN_EUR_unrel=1006;AN_EAS_unrel=1008;AN_AMR_unrel=694;AN_SAS_unrel=978;AN_AFR_unrel=1322;AF_EUR_unrel=0.548708;AF_EAS_unrel=0.550595;AF_AMR_unrel=0.573487;AF_SAS_unrel=0.611452;AF_AFR_unrel=0.33888;AC_EUR_unrel=552;AC_EAS_unrel=555;AC_AMR_unrel=398;AC_SAS_unrel=598;AC_AFR_unrel=448;AC_Het_EUR_unrel=228;AC_Het_EAS_unrel=251;AC_Het_AMR_unrel=176;AC_Het_SAS_unrel=234;AC_Het_AFR_unrel=296;AC_Hom_EUR_unrel=324;AC_Hom_EAS_unrel=304;AC_Hom_AMR_unrel=222;AC_Hom_SAS_unrel=364;AC_Hom_AFR_unrel=152;ANN=G\|intron_variant\|MODIFIER\|FGFR2\|ENSG00000066468\|transcript\|ENST00000457416.6\|protein_coding\|2/17\|c.110-12117T>C\|\|\|\|\|\|;1000Gp3_RA_GF=0.473243;1000Gp3_RR_GF=0.253994;1000Gp3_HomC=636,683;1000Gp3_AA_GF=0.272764;1000Gp3_AF=0.509385;AR_GENE=FGFR2;Entrez_gene_id=2263;RVIS=-1.50462206;RVIS_percentile=3.568058504;GDI=593.43974;GDI-Phred=4.93642;LoFtool_score=1.79E-03;Essential_gene=E;MGI_mouse_gene=Fgfr2;ZFIN_zebrafish_gene=fgfr2;ZFIN_zebrafish_phenotype_tag=abnormal;dbSNPBuildID=101                                                                                                                                                                  |
| Breast cancer (hormonal receptor positive   breast cancer) | 10    | 1.22E+08 | NA      | rs2981579  | NA               | A>G           | NA        | 3.60E-31  | GRASP        | 1491780 | PASS   | AC=3227;AF=0.503904;AN=6404;BaseQRankSum=0.098;ClippingRankSum=-0.024;DP=95506;FS=0;InbreedingCoeff=0.0499;MLEAC=3227;MLEAF=0.504;MQ=60;MQ0=0;MQRankSum=0.027;POSITIVE_TRAIN_SITE;QD=21.12;ReadPosRankSum=0.349;SOR=0.676;VQSLOD=22.69;culprit=MQ;AN_EUR=1266;AN_EAS=1170;AN_AMR=980;AN_SAS=1202;AN_AFR=1786;AF_EUR=0.550553;AF_EAS=0.549573;AF_AMR=0.568367;AF_SAS=0.610649;AF_AFR=0.333707;AC_EUR=697;AC_EAS=643;AC_AMR=557;AC_SAS=734;AC_AFR=596;AC_Het_EUR=293;AC_Het_EAS=283;AC_Het_AMR=249;AC_Het_SAS=296;AC_Het_AFR=400;AC_Het=1521;AC_Hom_EUR=404;AC_Hom_EAS=360;AC_Hom_AMR=308;AC_Hom_SAS=438;AC_Hom_AFR=196;AC_Hom=1706;HWE_EUR=0.108258;ExcHet_EUR=0.95804;HWE_EAS=0.616227;ExcHet_EAS=0.744677;HWE_AMR=0.462052;ExcHet_AMR=0.249731;HWE_SAS=0.439899;ExcHet_SAS=0.220669;HWE_AFR=0.880403;ExcHet_AFR=0.45117;HWE=0.00469594;ExcHet=0.997943;ME=0;AN_EUR_unrel=1006;AN_EAS_unrel=1008;AN_AMR_unrel=694;AN_SAS_unrel=978;AN_AFR_unrel=1322;AF_EUR_unrel=0.548708;AF_EAS_unrel=0.550595;AF_AMR_unrel=0.573487;AF_SAS_unrel=0.611452;AF_AFR_unrel=0.33888;AC_EUR_unrel=552;AC_EAS_unrel=555;AC_AMR_unrel=398;AC_SAS_unrel=598;AC_AFR_unrel=448;AC_Het_EUR_unrel=228;AC_Het_EAS_unrel=251;AC_Het_AMR_unrel=176;AC_Het_SAS_unrel=234;AC_Het_AFR_unrel=296;AC_Hom_EUR_unrel=324;AC_Hom_EAS_unrel=304;AC_Hom_AMR_unrel=222;AC_Hom_SAS_unrel=364;AC_Hom_AFR_unrel=152;ANN=G\|intron_variant\|MODIFIER\|FGFR2\|ENSG00000066468\|transcript\|ENST00000457416.6\|protein_coding\|2/17\|c.110-12117T>C\|\|\|\|\|\|;1000Gp3_RA_GF=0.473243;1000Gp3_RR_GF=0.253994;1000Gp3_HomC=636,683;1000Gp3_AA_GF=0.272764;1000Gp3_AF=0.509385;AR_GENE=FGFR2;Entrez_gene_id=2263;RVIS=-1.50462206;RVIS_percentile=3.568058504;GDI=593.43974;GDI-Phred=4.93642;LoFtool_score=1.79E-03;Essential_gene=E;MGI_mouse_gene=Fgfr2;ZFIN_zebrafish_gene=fgfr2;ZFIN_zebrafish_phenotype_tag=abnormal;dbSNPBuildID=101                                                                                                                                                                  |
| Alzheimers_disease                                         | 6     | 41161514 | TREM2   | rs75932628 | missense_variant | C>T           | NA        | 2.00E-12  | GWAS Catalog | 7296.45 | PASS   | AC=14;AF=0.00218613;AN=6404;BaseQRankSum=0.155;ClippingRankSum=0.481;DP=111734;FS=1.098;MLEAC=14;MLEAF=0.002186;MQ=60;MQ0=0;MQRankSum=0.307;POSITIVE_TRAIN_SITE;QD=13.46;ReadPosRankSum=0.839;SOR=0.59;VQSLOD=16.39;culprit=MQ;AN_EUR=1266;AN_EAS=1170;AN_AMR=980;AN_SAS=1202;AN_AFR=1786;AF_EUR=0.00631912;AF_EAS=0;AF_AMR=0.00306122;AF_SAS=0.00249584;AF_AFR=0;AC_EUR=8;AC_EAS=0;AC_AMR=3;AC_SAS=3;AC_AFR=0;AC_Het_EUR=8;AC_Het_EAS=0;AC_Het_AMR=3;AC_Het_SAS=3;AC_Het_AFR=0;AC_Het=14;AC_Hom_EUR=0;AC_Hom_EAS=0;AC_Hom_AMR=0;AC_Hom_SAS=0;AC_Hom_AFR=0;AC_Hom=0;HWE_EUR=1;ExcHet_EUR=0.977997;HWE_EAS=1;ExcHet_EAS=1;HWE_AMR=1;ExcHet_AMR=0.996936;HWE_SAS=1;ExcHet_SAS=0.997502;HWE_AFR=1;ExcHet_AFR=1;HWE=1;ExcHet=0.985861;ME=0;AN_EUR_unrel=1006;AN_EAS_unrel=1008;AN_AMR_unrel=694;AN_SAS_unrel=978;AN_AFR_unrel=1322;AF_EUR_unrel=0.00497018;AF_EAS_unrel=0;AF_AMR_unrel=0.00432277;AF_SAS_unrel=0.00204499;AF_AFR_unrel=0;AC_EUR_unrel=5;AC_EAS_unrel=0;AC_AMR_unrel=3;AC_SAS_unrel=2;AC_AFR_unrel=0;AC_Het_EUR_unrel=5;AC_Het_EAS_unrel=0;AC_Het_AMR_unrel=3;AC_Het_SAS_unrel=2;AC_Het_AFR_unrel=0;AC_Hom_EUR_unrel=0;AC_Hom_EAS_unrel=0;AC_Hom_AMR_unrel=0;AC_Hom_SAS_unrel=0;AC_Hom_AFR_unrel=0;ANN=T\|missense_variant\|MODERATE\|TREM2\|ENSG00000095970\|transcript\|ENST00000373113.7\|protein_coding\|2/5\|c.140G>A\|p.Arg47His\|234/1044\|140/693\|47/230\|\|;1000Gp3_RA_GF=0.00399361;1000Gp3_RR_GF=0.996006;1000Gp3_HomC=2494,0;1000Gp3_AA_GF=0;1000Gp3_AF=0.00199681;CLNSIG=3;CLNDBN=not_specified;CLNDSDBID=CN169374;CLNDSDB=MedGen;COSMIC_CNT=1;SIFT_pred=D/T/T;Uniprot_id_Polyphen2=././TREM2_HUMAN;Uniprot_aapos_Polyphen2=47;Polyphen2_HDIV_pred=D;Polyphen2_HVAR_pred=D;MutationTaster_pred=D/D/D;MutationAssessor_pred=M;CADD_phred=33;GERP++_NR=5.51;GERP++_RS=4.56;phyloP20way_mammalian=0.935;phastCons20way_mammalian=0.997;Entrez_gene_id=54209;RVIS=0.483275131;RVIS_percentile=79.25218212;GDI=446.17961;GDI-Phred=4.39760;LoFtool_score=8.38E-01;Essential_gene=N;MGI_mouse_gene=Trem2;dbSNPBuildID=132;ExAC_AF=0.002043 |
| Alzheimers_disease(late_onset)                             | 6     | 41161514 | TREM2   | rs75932628 | missense_variant | C>T           | NA        | 5.00E-24  | GWAS Catalog | 7296.45 | PASS   | AC=14;AF=0.00218613;AN=6404;BaseQRankSum=0.155;ClippingRankSum=0.481;DP=111734;FS=1.098;MLEAC=14;MLEAF=0.002186;MQ=60;MQ0=0;MQRankSum=0.307;POSITIVE_TRAIN_SITE;QD=13.46;ReadPosRankSum=0.839;SOR=0.59;VQSLOD=16.39;culprit=MQ;AN_EUR=1266;AN_EAS=1170;AN_AMR=980;AN_SAS=1202;AN_AFR=1786;AF_EUR=0.00631912;AF_EAS=0;AF_AMR=0.00306122;AF_SAS=0.00249584;AF_AFR=0;AC_EUR=8;AC_EAS=0;AC_AMR=3;AC_SAS=3;AC_AFR=0;AC_Het_EUR=8;AC_Het_EAS=0;AC_Het_AMR=3;AC_Het_SAS=3;AC_Het_AFR=0;AC_Het=14;AC_Hom_EUR=0;AC_Hom_EAS=0;AC_Hom_AMR=0;AC_Hom_SAS=0;AC_Hom_AFR=0;AC_Hom=0;HWE_EUR=1;ExcHet_EUR=0.977997;HWE_EAS=1;ExcHet_EAS=1;HWE_AMR=1;ExcHet_AMR=0.996936;HWE_SAS=1;ExcHet_SAS=0.997502;HWE_AFR=1;ExcHet_AFR=1;HWE=1;ExcHet=0.985861;ME=0;AN_EUR_unrel=1006;AN_EAS_unrel=1008;AN_AMR_unrel=694;AN_SAS_unrel=978;AN_AFR_unrel=1322;AF_EUR_unrel=0.00497018;AF_EAS_unrel=0;AF_AMR_unrel=0.00432277;AF_SAS_unrel=0.00204499;AF_AFR_unrel=0;AC_EUR_unrel=5;AC_EAS_unrel=0;AC_AMR_unrel=3;AC_SAS_unrel=2;AC_AFR_unrel=0;AC_Het_EUR_unrel=5;AC_Het_EAS_unrel=0;AC_Het_AMR_unrel=3;AC_Het_SAS_unrel=2;AC_Het_AFR_unrel=0;AC_Hom_EUR_unrel=0;AC_Hom_EAS_unrel=0;AC_Hom_AMR_unrel=0;AC_Hom_SAS_unrel=0;AC_Hom_AFR_unrel=0;ANN=T\|missense_variant\|MODERATE\|TREM2\|ENSG00000095970\|transcript\|ENST00000373113.7\|protein_coding\|2/5\|c.140G>A\|p.Arg47His\|234/1044\|140/693\|47/230\|\|;1000Gp3_RA_GF=0.00399361;1000Gp3_RR_GF=0.996006;1000Gp3_HomC=2494,0;1000Gp3_AA_GF=0;1000Gp3_AF=0.00199681;CLNSIG=3;CLNDBN=not_specified;CLNDSDBID=CN169374;CLNDSDB=MedGen;COSMIC_CNT=1;SIFT_pred=D/T/T;Uniprot_id_Polyphen2=././TREM2_HUMAN;Uniprot_aapos_Polyphen2=47;Polyphen2_HDIV_pred=D;Polyphen2_HVAR_pred=D;MutationTaster_pred=D/D/D;MutationAssessor_pred=M;CADD_phred=33;GERP++_NR=5.51;GERP++_RS=4.56;phyloP20way_mammalian=0.935;phastCons20way_mammalian=0.997;Entrez_gene_id=54209;RVIS=0.483275131;RVIS_percentile=79.25218212;GDI=446.17961;GDI-Phred=4.39760;LoFtool_score=8.38E-01;Essential_gene=N;MGI_mouse_gene=Trem2;dbSNPBuildID=132;ExAC_AF=0.002043 |
|                                                            | 8     | 95041772 | NDUFAF6 | rs7818382  | intron_variant   | C>T           | NA        | 8.00E-08  | GWAS Catalog | 1587730 | Pass   | AC=3017;AF=0.471112;AN=6404;BaseQRankSum=0.223;ClippingRankSum=0;DP=106075;FS=0;InbreedingCoeff=0.1044;MLEAC=3017;MLEAF=0.471;MQ=60;MQ0=0;MQRankSum=-0.022;POSITIVE_TRAIN_SITE;QD=21.26;ReadPosRankSum=0.319;SOR=0.686;VQSLOD=23.35;culprit=MQ;AN_EUR=1266;AN_EAS=1170;AN_AMR=980;AN_SAS=1202;AN_AFR=1786;AF_EUR=0.470774;AF_EAS=0.670085;AF_AMR=0.529592;AF_SAS=0.569884;AF_AFR=0.242441;AC_EUR=596;AC_EAS=784;AC_AMR=519;AC_SAS=685;AC_AFR=433;AC_Het_EUR=320;AC_Het_EAS=240;AC_Het_AMR=237;AC_Het_SAS=295;AC_Het_AFR=337;AC_Het=1429;AC_Hom_EUR=276;AC_Hom_EAS=544;AC_Hom_AMR=282;AC_Hom_SAS=390;AC_Hom_AFR=96;AC_Hom=1588;HWE_EUR=0.749959;ExcHet_EUR=0.395209;HWE_EAS=0.0919533;ExcHet_EAS=0.968002;HWE_AMR=0.526092;ExcHet_AMR=0.776731;HWE_SAS=1;ExcHet_SAS=0.529247;HWE_AFR=0.466134;ExcHet_AFR=0.239997;HWE=3.8034e-09;ExcHet=1;ME=0;AN_EUR_unrel=1006;AN_EAS_unrel=1008;AN_AMR_unrel=694;AN_SAS_unrel=978;AN_AFR_unrel=1322;AF_EUR_unrel=0.464215;AF_EAS_unrel=0.686508;AF_AMR_unrel=0.51585;AF_SAS_unrel=0.566462;AF_AFR_unrel=0.249622;AC_EUR_unrel=467;AC_EAS_unrel=692;AC_AMR_unrel=358;AC_SAS_unrel=554;AC_AFR_unrel=330;AC_Het_EUR_unrel=253;AC_Het_EAS_unrel=200;AC_Het_AMR_unrel=162;AC_Het_SAS_unrel=238;AC_Het_AFR_unrel=256;AC_Hom_EUR_unrel=214;AC_Hom_EAS_unrel=492;AC_Hom_AMR_unrel=196;AC_Hom_SAS_unrel=316;AC_Hom_AFR_unrel=74;ANN=T\|intron_variant\|MODIFIER\|NDUFAF6\|ENSG00000156170\|transcript\|ENST00000396124.8\|protein_coding\|4/8\|c.477+146C>T\|\|\|\|\|\|;1000Gp3_RA_GF=0.442891;1000Gp3_RR_GF=0.299121;1000Gp3_HomC=749,646;1000Gp3_AA_GF=0.257987;1000Gp3_AF=0.479433;Entrez_gene_id=137682;RVIS=-0.161524709;RVIS_percentile=41.6430762;GDI=43.71035;GDI-Phred=1.25843;MGI_mouse_gene=Ndufaf6;dbSNPBuildID=116                                                                                                                                                                                                                                                                                                      |
| Breast_cancer                                              | 10    | 1.22E+08 | FGFR2   | rs2981579  | intron_variant   | A>G           | NA        | 2.00E-170 | GWAS Catalog | 1491780 | PASS   | AC=3227;AF=0.503904;AN=6404;BaseQRankSum=0.098;ClippingRankSum=-0.024;DP=95506;FS=0;InbreedingCoeff=0.0499;MLEAC=3227;MLEAF=0.504;MQ=60;MQ0=0;MQRankSum=0.027;POSITIVE_TRAIN_SITE;QD=21.12;ReadPosRankSum=0.349;SOR=0.676;VQSLOD=22.69;culprit=MQ;AN_EUR=1266;AN_EAS=1170;AN_AMR=980;AN_SAS=1202;AN_AFR=1786;AF_EUR=0.550553;AF_EAS=0.549573;AF_AMR=0.568367;AF_SAS=0.610649;AF_AFR=0.333707;AC_EUR=697;AC_EAS=643;AC_AMR=557;AC_SAS=734;AC_AFR=596;AC_Het_EUR=293;AC_Het_EAS=283;AC_Het_AMR=249;AC_Het_SAS=296;AC_Het_AFR=400;AC_Het=1521;AC_Hom_EUR=404;AC_Hom_EAS=360;AC_Hom_AMR=308;AC_Hom_SAS=438;AC_Hom_AFR=196;AC_Hom=1706;HWE_EUR=0.108258;ExcHet_EUR=0.95804;HWE_EAS=0.616227;ExcHet_EAS=0.744677;HWE_AMR=0.462052;ExcHet_AMR=0.249731;HWE_SAS=0.439899;ExcHet_SAS=0.220669;HWE_AFR=0.880403;ExcHet_AFR=0.45117;HWE=0.00469594;ExcHet=0.997943;ME=0;AN_EUR_unrel=1006;AN_EAS_unrel=1008;AN_AMR_unrel=694;AN_SAS_unrel=978;AN_AFR_unrel=1322;AF_EUR_unrel=0.548708;AF_EAS_unrel=0.550595;AF_AMR_unrel=0.573487;AF_SAS_unrel=0.611452;AF_AFR_unrel=0.33888;AC_EUR_unrel=552;AC_EAS_unrel=555;AC_AMR_unrel=398;AC_SAS_unrel=598;AC_AFR_unrel=448;AC_Het_EUR_unrel=228;AC_Het_EAS_unrel=251;AC_Het_AMR_unrel=176;AC_Het_SAS_unrel=234;AC_Het_AFR_unrel=296;AC_Hom_EUR_unrel=324;AC_Hom_EAS_unrel=304;AC_Hom_AMR_unrel=222;AC_Hom_SAS_unrel=364;AC_Hom_AFR_unrel=152;ANN=G\|intron_variant\|MODIFIER\|FGFR2\|ENSG00000066468\|transcript\|ENST00000457416.6\|protein_coding\|2/17\|c.110-12117T>C\|\|\|\|\|\|;1000Gp3_RA_GF=0.473243;1000Gp3_RR_GF=0.253994;1000Gp3_HomC=636,683;1000Gp3_AA_GF=0.272764;1000Gp3_AF=0.509385;AR_GENE=FGFR2;Entrez_gene_id=2263;RVIS=-1.50462206;RVIS_percentile=3.568058504;GDI=593.43974;GDI-Phred=4.93642;LoFtool_score=1.79E-03;Essential_gene=E;MGI_mouse_gene=Fgfr2;ZFIN_zebrafish_gene=fgfr2;ZFIN_zebrafish_phenotype_tag=abnormal;dbSNPBuildID=101                                                                                                                                                                  |
|                                                            | 10    | 1.22E+08 | NR      | rs2981579  | intron_variant   | A>G           | NA        | 6.00E-164 | GWAS Catalog | 1491780 | PASS   | AC=3227;AF=0.503904;AN=6404;BaseQRankSum=0.098;ClippingRankSum=-0.024;DP=95506;FS=0;InbreedingCoeff=0.0499;MLEAC=3227;MLEAF=0.504;MQ=60;MQ0=0;MQRankSum=0.027;POSITIVE_TRAIN_SITE;QD=21.12;ReadPosRankSum=0.349;SOR=0.676;VQSLOD=22.69;culprit=MQ;AN_EUR=1266;AN_EAS=1170;AN_AMR=980;AN_SAS=1202;AN_AFR=1786;AF_EUR=0.550553;AF_EAS=0.549573;AF_AMR=0.568367;AF_SAS=0.610649;AF_AFR=0.333707;AC_EUR=697;AC_EAS=643;AC_AMR=557;AC_SAS=734;AC_AFR=596;AC_Het_EUR=293;AC_Het_EAS=283;AC_Het_AMR=249;AC_Het_SAS=296;AC_Het_AFR=400;AC_Het=1521;AC_Hom_EUR=404;AC_Hom_EAS=360;AC_Hom_AMR=308;AC_Hom_SAS=438;AC_Hom_AFR=196;AC_Hom=1706;HWE_EUR=0.108258;ExcHet_EUR=0.95804;HWE_EAS=0.616227;ExcHet_EAS=0.744677;HWE_AMR=0.462052;ExcHet_AMR=0.249731;HWE_SAS=0.439899;ExcHet_SAS=0.220669;HWE_AFR=0.880403;ExcHet_AFR=0.45117;HWE=0.00469594;ExcHet=0.997943;ME=0;AN_EUR_unrel=1006;AN_EAS_unrel=1008;AN_AMR_unrel=694;AN_SAS_unrel=978;AN_AFR_unrel=1322;AF_EUR_unrel=0.548708;AF_EAS_unrel=0.550595;AF_AMR_unrel=0.573487;AF_SAS_unrel=0.611452;AF_AFR_unrel=0.33888;AC_EUR_unrel=552;AC_EAS_unrel=555;AC_AMR_unrel=398;AC_SAS_unrel=598;AC_AFR_unrel=448;AC_Het_EUR_unrel=228;AC_Het_EAS_unrel=251;AC_Het_AMR_unrel=176;AC_Het_SAS_unrel=234;AC_Het_AFR_unrel=296;AC_Hom_EUR_unrel=324;AC_Hom_EAS_unrel=304;AC_Hom_AMR_unrel=222;AC_Hom_SAS_unrel=364;AC_Hom_AFR_unrel=152;ANN=G\|intron_variant\|MODIFIER\|FGFR2\|ENSG00000066468\|transcript\|ENST00000457416.6\|protein_coding\|2/17\|c.110-12117T>C\|\|\|\|\|\|;1000Gp3_RA_GF=0.473243;1000Gp3_RR_GF=0.253994;1000Gp3_HomC=636,683;1000Gp3_AA_GF=0.272764;1000Gp3_AF=0.509385;AR_GENE=FGFR2;Entrez_gene_id=2263;RVIS=-1.50462206;RVIS_percentile=3.568058504;GDI=593.43974;GDI-Phred=4.93642;LoFtool_score=1.79E-03;Essential_gene=E;MGI_mouse_gene=Fgfr2;ZFIN_zebrafish_gene=fgfr2;ZFIN_zebrafish_phenotype_tag=abnormal;dbSNPBuildID=101                                                                                                                                                                  |
|                                                            | 10    | 1.22E+08 | FGFR2   | rs2981579  | intron_variant   | A>G           | NA        | 4.00E-31  | GWAS Catalog | 1491780 | PASS   | AC=3227;AF=0.503904;AN=6404;BaseQRankSum=0.098;ClippingRankSum=-0.024;DP=95506;FS=0;InbreedingCoeff=0.0499;MLEAC=3227;MLEAF=0.504;MQ=60;MQ0=0;MQRankSum=0.027;POSITIVE_TRAIN_SITE;QD=21.12;ReadPosRankSum=0.349;SOR=0.676;VQSLOD=22.69;culprit=MQ;AN_EUR=1266;AN_EAS=1170;AN_AMR=980;AN_SAS=1202;AN_AFR=1786;AF_EUR=0.550553;AF_EAS=0.549573;AF_AMR=0.568367;AF_SAS=0.610649;AF_AFR=0.333707;AC_EUR=697;AC_EAS=643;AC_AMR=557;AC_SAS=734;AC_AFR=596;AC_Het_EUR=293;AC_Het_EAS=283;AC_Het_AMR=249;AC_Het_SAS=296;AC_Het_AFR=400;AC_Het=1521;AC_Hom_EUR=404;AC_Hom_EAS=360;AC_Hom_AMR=308;AC_Hom_SAS=438;AC_Hom_AFR=196;AC_Hom=1706;HWE_EUR=0.108258;ExcHet_EUR=0.95804;HWE_EAS=0.616227;ExcHet_EAS=0.744677;HWE_AMR=0.462052;ExcHet_AMR=0.249731;HWE_SAS=0.439899;ExcHet_SAS=0.220669;HWE_AFR=0.880403;ExcHet_AFR=0.45117;HWE=0.00469594;ExcHet=0.997943;ME=0;AN_EUR_unrel=1006;AN_EAS_unrel=1008;AN_AMR_unrel=694;AN_SAS_unrel=978;AN_AFR_unrel=1322;AF_EUR_unrel=0.548708;AF_EAS_unrel=0.550595;AF_AMR_unrel=0.573487;AF_SAS_unrel=0.611452;AF_AFR_unrel=0.33888;AC_EUR_unrel=552;AC_EAS_unrel=555;AC_AMR_unrel=398;AC_SAS_unrel=598;AC_AFR_unrel=448;AC_Het_EUR_unrel=228;AC_Het_EAS_unrel=251;AC_Het_AMR_unrel=176;AC_Het_SAS_unrel=234;AC_Het_AFR_unrel=296;AC_Hom_EUR_unrel=324;AC_Hom_EAS_unrel=304;AC_Hom_AMR_unrel=222;AC_Hom_SAS_unrel=364;AC_Hom_AFR_unrel=152;ANN=G\|intron_variant\|MODIFIER\|FGFR2\|ENSG00000066468\|transcript\|ENST00000457416.6\|protein_coding\|2/17\|c.110-12117T>C\|\|\|\|\|\|;1000Gp3_RA_GF=0.473243;1000Gp3_RR_GF=0.253994;1000Gp3_HomC=636,683;1000Gp3_AA_GF=0.272764;1000Gp3_AF=0.509385;AR_GENE=FGFR2;Entrez_gene_id=2263;RVIS=-1.50462206;RVIS_percentile=3.568058504;GDI=593.43974;GDI-Phred=4.93642;LoFtool_score=1.79E-03;Essential_gene=E;MGI_mouse_gene=Fgfr2;ZFIN_zebrafish_gene=fgfr2;ZFIN_zebrafish_phenotype_tag=abnormal;dbSNPBuildID=101                                                                                                                                                                  |
|                                                            | 10    | 1.22E+08 |         | rs2981579  | intron_variant   | A>G           | NA        | 3.00E-11  | GWAS Catalog | 1491780 | PASS   | AC=3227;AF=0.503904;AN=6404;BaseQRankSum=0.098;ClippingRankSum=-0.024;DP=95506;FS=0;InbreedingCoeff=0.0499;MLEAC=3227;MLEAF=0.504;MQ=60;MQ0=0;MQRankSum=0.027;POSITIVE_TRAIN_SITE;QD=21.12;ReadPosRankSum=0.349;SOR=0.676;VQSLOD=22.69;culprit=MQ;AN_EUR=1266;AN_EAS=1170;AN_AMR=980;AN_SAS=1202;AN_AFR=1786;AF_EUR=0.550553;AF_EAS=0.549573;AF_AMR=0.568367;AF_SAS=0.610649;AF_AFR=0.333707;AC_EUR=697;AC_EAS=643;AC_AMR=557;AC_SAS=734;AC_AFR=596;AC_Het_EUR=293;AC_Het_EAS=283;AC_Het_AMR=249;AC_Het_SAS=296;AC_Het_AFR=400;AC_Het=1521;AC_Hom_EUR=404;AC_Hom_EAS=360;AC_Hom_AMR=308;AC_Hom_SAS=438;AC_Hom_AFR=196;AC_Hom=1706;HWE_EUR=0.108258;ExcHet_EUR=0.95804;HWE_EAS=0.616227;ExcHet_EAS=0.744677;HWE_AMR=0.462052;ExcHet_AMR=0.249731;HWE_SAS=0.439899;ExcHet_SAS=0.220669;HWE_AFR=0.880403;ExcHet_AFR=0.45117;HWE=0.00469594;ExcHet=0.997943;ME=0;AN_EUR_unrel=1006;AN_EAS_unrel=1008;AN_AMR_unrel=694;AN_SAS_unrel=978;AN_AFR_unrel=1322;AF_EUR_unrel=0.548708;AF_EAS_unrel=0.550595;AF_AMR_unrel=0.573487;AF_SAS_unrel=0.611452;AF_AFR_unrel=0.33888;AC_EUR_unrel=552;AC_EAS_unrel=555;AC_AMR_unrel=398;AC_SAS_unrel=598;AC_AFR_unrel=448;AC_Het_EUR_unrel=228;AC_Het_EAS_unrel=251;AC_Het_AMR_unrel=176;AC_Het_SAS_unrel=234;AC_Het_AFR_unrel=296;AC_Hom_EUR_unrel=324;AC_Hom_EAS_unrel=304;AC_Hom_AMR_unrel=222;AC_Hom_SAS_unrel=364;AC_Hom_AFR_unrel=152;ANN=G\|intron_variant\|MODIFIER\|FGFR2\|ENSG00000066468\|transcript\|ENST00000457416.6\|protein_coding\|2/17\|c.110-12117T>C\|\|\|\|\|\|;1000Gp3_RA_GF=0.473243;1000Gp3_RR_GF=0.253994;1000Gp3_HomC=636,683;1000Gp3_AA_GF=0.272764;1000Gp3_AF=0.509385;AR_GENE=FGFR2;Entrez_gene_id=2263;RVIS=-1.50462206;RVIS_percentile=3.568058504;GDI=593.43974;GDI-Phred=4.93642;LoFtool_score=1.79E-03;Essential_gene=E;MGI_mouse_gene=Fgfr2;ZFIN_zebrafish_gene=fgfr2;ZFIN_zebrafish_phenotype_tag=abnormal;dbSNPBuildID=101                                                                                                                                                                  |
|                                                            | 10    | 1.22E+08 | FGFR2   | rs2981579  | intron_variant   | A>G           | NA        | 2.00E-10  | GWAS Catalog | 1491780 | PASS   | AC=3227;AF=0.503904;AN=6404;BaseQRankSum=0.098;ClippingRankSum=-0.024;DP=95506;FS=0;InbreedingCoeff=0.0499;MLEAC=3227;MLEAF=0.504;MQ=60;MQ0=0;MQRankSum=0.027;POSITIVE_TRAIN_SITE;QD=21.12;ReadPosRankSum=0.349;SOR=0.676;VQSLOD=22.69;culprit=MQ;AN_EUR=1266;AN_EAS=1170;AN_AMR=980;AN_SAS=1202;AN_AFR=1786;AF_EUR=0.550553;AF_EAS=0.549573;AF_AMR=0.568367;AF_SAS=0.610649;AF_AFR=0.333707;AC_EUR=697;AC_EAS=643;AC_AMR=557;AC_SAS=734;AC_AFR=596;AC_Het_EUR=293;AC_Het_EAS=283;AC_Het_AMR=249;AC_Het_SAS=296;AC_Het_AFR=400;AC_Het=1521;AC_Hom_EUR=404;AC_Hom_EAS=360;AC_Hom_AMR=308;AC_Hom_SAS=438;AC_Hom_AFR=196;AC_Hom=1706;HWE_EUR=0.108258;ExcHet_EUR=0.95804;HWE_EAS=0.616227;ExcHet_EAS=0.744677;HWE_AMR=0.462052;ExcHet_AMR=0.249731;HWE_SAS=0.439899;ExcHet_SAS=0.220669;HWE_AFR=0.880403;ExcHet_AFR=0.45117;HWE=0.00469594;ExcHet=0.997943;ME=0;AN_EUR_unrel=1006;AN_EAS_unrel=1008;AN_AMR_unrel=694;AN_SAS_unrel=978;AN_AFR_unrel=1322;AF_EUR_unrel=0.548708;AF_EAS_unrel=0.550595;AF_AMR_unrel=0.573487;AF_SAS_unrel=0.611452;AF_AFR_unrel=0.33888;AC_EUR_unrel=552;AC_EAS_unrel=555;AC_AMR_unrel=398;AC_SAS_unrel=598;AC_AFR_unrel=448;AC_Het_EUR_unrel=228;AC_Het_EAS_unrel=251;AC_Het_AMR_unrel=176;AC_Het_SAS_unrel=234;AC_Het_AFR_unrel=296;AC_Hom_EUR_unrel=324;AC_Hom_EAS_unrel=304;AC_Hom_AMR_unrel=222;AC_Hom_SAS_unrel=364;AC_Hom_AFR_unrel=152;ANN=G\|intron_variant\|MODIFIER\|FGFR2\|ENSG00000066468\|transcript\|ENST00000457416.6\|protein_coding\|2/17\|c.110-12117T>C\|\|\|\|\|\|;1000Gp3_RA_GF=0.473243;1000Gp3_RR_GF=0.253994;1000Gp3_HomC=636,683;1000Gp3_AA_GF=0.272764;1000Gp3_AF=0.509385;AR_GENE=FGFR2;Entrez_gene_id=2263;RVIS=-1.50462206;RVIS_percentile=3.568058504;GDI=593.43974;GDI-Phred=4.93642;LoFtool_score=1.79E-03;Essential_gene=E;MGI_mouse_gene=Fgfr2;ZFIN_zebrafish_gene=fgfr2;ZFIN_zebrafish_phenotype_tag=abnormal;dbSNPBuildID=101                                                                                                                                                                  |

## Additional Resources

- The VCF format is a standard file format for storing genetic variation data, and it is widely used in bioinformatics research. More information on the VCF format can be found at the [VCF specification page](http://samtools.github.io/hts-specs/VCFv4.3.pdf)

[![CRAN_Status_Badge](http://www.r-pkg.org/badges/version/DisVar)](https://cran.r-project.org/package=DisVar)
