---
author: "Khunanon Chanasongkhram"
date: "07/07/2023"
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# DisVar: An R library for identifying variants associated with diseases using personal genetic information

<!-- badges: start -->
<!-- badges: end -->
DisVar is an R package that finds diseases from a VCF file by comparing the variants in the VCF file with those in various databases. The package currently supports five databases: GWASdb, GRASP, GWAS Catalog, GAD, and Johnson and O'Donnell.

## Features

- Finds diseases from a VCF file by comparing the variants in the VCF file with those in various databases.

- Supports five databases: GWASdb, GRASP, GWAS Catalog, GAD, and Johnson and O'Donnell.

## Installation

## Installing the Package in R Studio
You can install the development version of DisVar from [GitHub](https://github.com/). To install the package, you can use the `devtools` package and run the following command:

``` r
# install.packages("devtools")
devtools::install_github("Khunanon-Chanasongkhram/DisVar")
```

## Installing the Package on Linux servers

### Step 1: Installing Minoconda 3

To install Miniconda via the terminal, you can follow these steps:
1.Create a directory to install Miniconda into:

``` 
mkdir -p ~/miniconda3
```

2.Download the latest Miniconda3 based install script for Linux 64 bit:
``` 
wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda3/miniconda.sh
```

3.Run the install script:
``` 
bash ~/miniconda3/miniconda.sh -b -u -p ~/miniconda3
```

After then, you can restart your shell, and conda will be ready for use.

For further details regarding conda installation please see [Miniconda linux installers](https://docs.conda.io/en/latest/miniconda.html#linux-installers)

### Step 2: Installing R using conda
Create a new conda environment for R by running the following command:
``` 
conda create -n r-environment r-base
```

Please answer "y" for the question "Proceed ([y]/n)?".
This will create a new environment named 'r-environment' with R installed.
Now you can activate the new environment by running:
```
conda activate r-environment
```

If you want to and deactivate an active environment just running:

```
conda deactivate
```

### Step 3: Installing the DisVar Package
After activate r-environment via conda. Run R in Linux by following command:
```
R
```

Then install the package
```
remotes::install_github(repo = "Khunanon-Chanasongkhram/DisVar",dependencies = TRUE,upgrade = TRUE)
```
Now the DisVar is ready to use!

## Dependencies
The package depends on the following packages:

- `sqldf`
- `data.table`
- `dplyr`

## Usage

The main function in the package is `DisVar()`, which takes one arguments:

- `file`: the name of the VCF file.

The function returns a data frame containing variants associated with diseases compared from various databases.

### Examples

Here's an example of how to use the function:

```{r example}
library(DisVar)
DisVar("file_name.vcf")
```

You will get this output file: `file_name`_diseases_output.txt

When you test the program with the test file (`file_name.vcf`), the results are shown in the following table.

|Disease                                                 |Chr|Position|Gene      |Variant_id|Variant_type|Confident/P-value|Allele DB|Confident|DB          |
|--------------------------------------------------------|---|--------|----------|----------|------------|-----------------|---------|---------|------------|
|Breast cancer                                           |3  |4700592 |ITPR1     |rs6762644 |intron      |A>G              |A>G      |2e-12    |GWASdb      |
|                                                        |3  |4700592 |NA        |rs6762644 |NA          |A>G              |NA       |2.2e-12  |GRASP       |
|Breast cancer (Estrogen receptor positive breast cancer)|3  |4700592 |NA        |rs6762644 |NA          |A>G              |NA       |1.4e-08  |GRASP       |
|Breast cancer (invasive breast cancer)                  |3  |4700592 |NA        |rs6762644 |NA          |A>G              |NA       |1.2e-09  |GRASP       |
|Breast_cancer                                           |3  |4700592 |EGOT/ITPR1|rs6762644 |intron      |A>G              |NA       |4e-18    |GWAS Catalog|
|                                                        |3  |4700592 |ITPR1,EGOT|rs6762644 |intron      |A>G              |NA       |2e-12    |GWAS Catalog|
|                                                        |3  |4700592 |NR        |rs6762644 |intron      |A>G              |NA       |9e-12    |GWAS Catalog|


## Additional Resources

- The VCF format is a standard file format for storing genetic variation data, and it is widely used in bioinformatics research. More information on the VCF format can be found at the [VCF specification page](http://samtools.github.io/hts-specs/VCFv4.3.pdf)

[![CRAN_Status_Badge](http://www.r-pkg.org/badges/version/DisVar)](https://cran.r-project.org/package=DisVar)
