---
title: "DisVar: An R library for identifying variants associated with diseases using personal genetic information"
author: "Khunanon Chanasongkhram"
date: "11/01/2023"
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# DisVar: An R library for identifying variants associated with diseases using personal genetic information

<!-- badges: start -->
<!-- badges: end -->
DisVar is an R package that finds diseases from a VCF file by comparing the variants in the VCF file with those in various databases. The package currently supports five databases: GWASdb, GRASP, GWAS Catalog, GAD, and, Johnson and O'Donnell.


## Installation

You can install the development version of DisVar from [GitHub](https://github.com/) to install the package, you can use the `devtools` package and run the following command:

``` r
# install.packages("devtools")
devtools::install_github("Khunanon-Chanasongkhram/DisVar")
```
## Usage

The main function in the package is 'DisVar()', which takes two arguments:

• 'file': the name of the VCF file.
• 'skip': the number of lines to skip at the beginning of the VCF file. (default: 0)
The function returns a data frame containing variant information from various databases.

Here's an example of how to use the function:

```{r example}
library(DisVar)
DisVar("file_name.vcf", skip = 28)
```
The package also depends on two external packages 'sqldf' and 'data.table', they need to be installed before using the package.

