---
title: "Introduction to DisVar"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Introduction to DisVar}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# DisVar: An R library for identifying variants associated with diseases using personal genetic information

DisVar is an R package that finds diseases from a VCF file by comparing the variants in the VCF file with those in various databases. The package currently supports five databases: GWASdb, GRASP, GWAS Catalog, GAD, and Johnson and O'Donnell.

## Features

- Finds diseases from a VCF file by comparing the variants in the VCF file with those in various databases.

- Supports five databases: GWASdb, GRASP, GWAS Catalog, GAD, and Johnson and O'Donnell.

## Dependencies
The package depends on the following packages:

- `data.table`
- `dplyr`
- `sqldf`
- `stats`
- `utils`

## Usage

The main function in the package is `DisVar()`, which takes one arguments:

- `file`: the name of the VCF file.

The function returns a data frame containing variants associated with diseases compared from various databases.

### Examples

Here's an example of how to use the function:

```{r example}
library(DisVar)
DisVar("file_name.vcf")
```

You will get this output file: `file_name`_DisVar.tsv

When you test the program with the test file (`file_name.vcf`)

## Publication

This project has been published in the [PeerJ](https://peerj.com) journal.

You can access the full publication here:

[DisVar: an R library for identifying variants associated with diseases using large-scale personal genetic information](https://peerj.com/articles/16086)

[![DOI](https://img.shields.io/badge/DOI-10.7717/peerj.16086-blue)](https://doi.org/10.7717/peerj.16086)

Published: 28 September 2023

## Additional Resources

- The VCF format is a standard file format for storing genetic variation data, and it is widely used in bioinformatics research. More information on the VCF format can be found at the [VCF specification page](http://samtools.github.io/hts-specs/VCFv4.3.pdf)

[![DOI](https://zenodo.org/badge/587268783.svg)](https://zenodo.org/badge/latestdoi/587268783)

